packages/shared/src/components/squads/SharePostBar.tsx

```diff
@@ -30,23 +31,42 @@ function SharePostBar({
   onNewSquadPost,
   disabled = false,
 }: SharePostBarProps): ReactElement {
+  const { displayToast } = useToastNotification();
   const [url, setUrl] = useState('');
   const onSharedSuccessfully = () => setUrl('');
-  const { mutateAsync: getPost } = useMutation(getPostByUrl, {
-    onSuccess: (post) => {
-      onNewSquadPost({ post, onSharedSuccessfully });
-    },
-    onError: (err: ApiErrorResult, link) => {
-      if (
-        link === '' ||
-        allowedSubmissionErrors.includes(
-          err?.response?.errors?.[0].extensions.code,
-        )
-      ) {
-        onNewSquadPost({ url, onSharedSuccessfully });
-      }
+  const { mutateAsync: getPrivateLink, isLoading: isLoadingPreview } =
```

Comment:
This code is now duplicate right?
Consider moving it to a hook maybe for uniform behaviour.